CONTRIBUTING.md

```diff
@@ -2,18 +2,49 @@
 
 If you're reading this section, you're probably interested in contributing to Starfish.  Welcome and thanks for your interest in contributing!
 
-# Code of conduct
+# How can I contribute?
+Starfish is designed to specify pipeline recipes for image processing. To support this use, the library is composed as a series of `pipeline_component` modules.
+The objects that sub-class `pipeline_component` spawn a command-line interface that should identically track the API of the python library.
 
+A typical starfish run consists of running one or more image processing filter stages, and identifying features through either a spot- or pixel-based approach.
+The identified features are then decoded into the genes that they correspond to by mapping the fluorescence channel (and optionally hybridization round) using a codebook.
+Finally, the filtered data are segmented, identifying which cell each feature belongs to.
+
+Implementing a new module of an existing `pipeline_component` is a simple process. For example, to add another image filtering step, one would:
+
+1. Create a new python file `new_filter.py` in the `starfish/pipeline/filter/` directory.
+
+2. Create a new python class `NewFilter(PipelineComponent)` in the `new_filter.py` class.
```

Comment:
This, I believe, is incorrect.  You would want to subclass `FilterAlgorithmBase` or something akin to that.